src/main/java/edu/cornell/kfs/coa/businessobject/CuAccountGlobal.java

```diff
@@ -2,164 +2,824 @@
-                }
-    
-                // FINANCIAL HIGHER ED FUNCTION CODE
-                if (StringUtils.isNotBlank(financialHigherEdFunctionCd)) {
-                    account.setFinancialHigherEdFunctionCd(financialHigherEdFunctionCd);
-                }
-    
-                // SUFFICIENT FUNDS CODE
-                if (StringUtils.isNotBlank(accountSufficientFundsCode)) {
-                    account.setAccountSufficientFundsCode(accountSufficientFundsCode);
-                }
-    
-                // LABOR BENEFIT RATE CATEGORY CODE
-                if (StringUtils.isNotBlank(getLaborBenefitRateCategoryCode())) {
-                    account.setLaborBenefitRateCategoryCode(getLaborBenefitRateCategoryCode());
-                }
-    
-                // PENDING ACCOUNT SUFFICIENT FUNDS CODE INDICATOR
-                if (pendingAcctSufficientFundsIndicator != null) {
-                    account.setPendingAcctSufficientFundsIndicator(pendingAcctSufficientFundsIndicator);
-                }
-    
-                // MAJOR REPORTING CATEGORY CODE 
-                if (StringUtils.isNotBlank(majorReportingCategoryCode)) {
-                    ((AccountExtendedAttribute) account.getExtension()).setMajorReportingCategoryCode(majorReportingCategoryCode);
-                }
-                
-                
-                persistables.add(account);
+    public CuAccountGlobal() {
+    	super();
+    	indirectCostRecoveryAccounts = new ArrayList<IndirectCostRecoveryAccountChange>();
+	}
+
     
-            }
-        }
+	@Override
+    public List<PersistableBusinessObject> generateGlobalChangesToPersist() {
+        List<PersistableBusinessObject> persistables = new ArrayList<PersistableBusinessObject>();
 
+        for (AccountGlobalDetail accountGlobalDetail : accountGlobalDetails) {
+        	updateAccountValuesAndAddToPersistablesList(persistables, accountGlobalDetail);
+        }
+        
         return persistables;
     }
+
+	private void updateAccountValuesAndAddToPersistablesList(
+			List<PersistableBusinessObject> persistables,
+			AccountGlobalDetail accountGlobalDetail) {
+		Account account = updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccountGlobalDoc(accountGlobalDetail);
+
+		if (ObjectUtils.isNotNull(account)) {
+			persistables.add(account);
+		}
+	}
+    
+	private Account updateAccountValuesOnAccountGlobalDetailWithNewValuesFromAccountGlobalDoc(
+			GlobalBusinessObjectDetailBase globalDetail) {
+		AccountGlobalDetail accountGlobalDetail = (AccountGlobalDetail) globalDetail;
+		Account account = SpringContext.getBean(BusinessObjectService.class).findByPrimaryKey(Account.class,accountGlobalDetail.getPrimaryKeys());
+
+		if (ObjectUtils.isNotNull(account)) {
+			updateAccountBasicFields(account);
+			updateAccountExtendedAttribute(account);
+			
+			if(ObjectUtils.isNotNull(this.getIndirectCostRecoveryAccounts()) && this.getIndirectCostRecoveryAccounts().size() > 0){
```

Comment:
since account.getIndirectCostRecoveryAccounts() is being passed to the updateIcrAccounts method, should it be what is checked in this if statement?